maven-core/src/test/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactoryTest.java

```diff
@@ -0,0 +1,401 @@
+
+        DefaultRepositorySystemSessionFactory systemSessionFactory = new DefaultRepositorySystemSessionFactory(
+                artifactHandlerManager,
+                aetherRepositorySystem,
+                null,
+                settingsDecrypter,
+                eventSpyDispatcher,
+                mavenRepositorySystem,
+                information);
+
+        int requestTimeout = (Integer) systemSessionFactory
+                .newRepositorySession(request)
+                .getConfigProperties()
+                .get(ConfigurationProperties.REQUEST_TIMEOUT + "." + server.getId());
+        assertEquals(3000, requestTimeout);
+    }
+
+    @Test
+    void readTimeoutFromHttpConfigurationTest() throws InvalidRepositoryException {
+        Server server = new Server();
+        server.setId("repository");
+        server.setUsername("jason");
+        server.setPassword("abc123");
+
+        Xpp3Dom configuration = new Xpp3Dom("configuration");
+        Xpp3Dom httpConfiguration = new Xpp3Dom("httpConfiguration");
+        Xpp3Dom all = new Xpp3Dom("all");
+        Xpp3Dom readTimeoutConfiguration = new Xpp3Dom("readTimeout");
+        readTimeoutConfiguration.setValue("3000");
+
+        all.addChild(readTimeoutConfiguration);
+        httpConfiguration.addChild(all);
+        configuration.addChild(httpConfiguration);
+
+        server.setConfiguration(configuration);
+
+        MavenExecutionRequest request = new DefaultMavenExecutionRequest();
+        request.setLocalRepository(getLocalRepository());
+        List<Server> servers = new ArrayList<>();
+        servers.add(server);
+        request.setServers(servers);
+
+        DefaultRepositorySystemSessionFactory systemSessionFactory = new DefaultRepositorySystemSessionFactory(
+                artifactHandlerManager,
+                aetherRepositorySystem,
+                null,
+                settingsDecrypter,
+                eventSpyDispatcher,
+                mavenRepositorySystem,
+                information);
+
+        int requestTimeout = (Integer) systemSessionFactory
+                .newRepositorySession(request)
+                .getConfigProperties()
+                .get(ConfigurationProperties.REQUEST_TIMEOUT + "." + server.getId());
+        assertEquals(3000, requestTimeout);
+    }
+
+    @Test
+    void transportConfigurationTest() throws InvalidRepositoryException {
+        DefaultRepositorySystemSessionFactory systemSessionFactory = new DefaultRepositorySystemSessionFactory(
+                artifactHandlerManager,
+                aetherRepositorySystem,
+                null,
+                settingsDecrypter,
+                eventSpyDispatcher,
+                mavenRepositorySystem,
+                information);
+
+        MavenExecutionRequest request = new DefaultMavenExecutionRequest();
+        request.setLocalRepository(getLocalRepository());
+
+        // native
+        System.setProperty("maven.resolver.transport", "native");
```

Comment:
I would like to not change system properties - it can have impact on other test.
We can simply copy system properties to own instance and use it for test.
Maybe we need only one property for this test?

Reply:
@slawekjaranowski 
Agreed. I have revised this part.